tests/test_core.py

```diff
@@ -597,6 +598,7 @@ async def modify_dir(directory):
         with TemporaryDirectory() as directory:
             await asyncio.gather(watch_dirs([directory]), modify_dir(directory))
 
+    @pytest.mark.xfail(reason="Test seems to fail on Mac tests - should investigate")
```

Comment:
This test was xfailed in #1599 but we might have removed it, I couldn't find anything about this test being fixed in MacOS so I've just xfailed it again